messenger/message-recorder.rst

```diff
@@ -0,0 +1,77 @@
+.. index::
+    single: Messenger; Record messages
+
+Events Recorder: Handle Events After CommandHandler Is Done
+===========================================================
+
+Let's take the example of an application that has a command (a CQRS message) named
+``CreateUser``. That command is handled by the ``CreateUserHandler`` which creates
+a ``User`` object, stores that object to a database and dispatches a ``UserCreated`` event.
+That event is also a normal message but is handled by an *event* bus.
+
+There are many subscribers to the ``UserCreated`` event, one subscriber may send
+a welcome email to the new user. We are using the ``DoctrineTransactionMiddleware``
+to wrap all database queries in one database transaction.
+
+**Problem:** If an exception is thrown when sending the welcome email, then the user
+will not be created because the ``DoctrineTransactionMiddleware`` will rollback the
+Doctrine transaction, in which the user has been created.
+
+**Solution:** The solution is to not dispatch the ``UserCreated`` event in the
+``CreateUserHandler`` but to just "record" the events. The recorded events will
+be dispatched after ``DoctrineTransactionMiddleware`` has committed the transaction.
+
+To enable this, you simply just add the ``messenger.middleware.handles_recorded_messages``
```

Comment:
We should remove "simply just", because it's just filler, and not everyone would consider this simple.